src/adapters/ws/client.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * WS client adapter
+ */
+import Adapter from '../../lib/adapter.js';
+import GleeMessage from '../../lib/message.js';
+import ws from 'ws';
+
+
+class WsClientAdapter extends Adapter {
+    private client: ws
+
+    name(): string {
+        return 'WS adapter';
+    }
+
+    async connect(): Promise<this> {
+        return this._connect();
+    }
+
+    async send(message: GleeMessage) {
+        return this._send(message);
+    }
+
+    _connect(): Promise<this> {
```

Comment:
You may want to make this private as well. When I did the first adapters, JS didn't support private stuff on classes 😄